### PR TITLE
Guides getting started

### PR DESCRIPTION
**Pull request by elenatorro**
Related with https://github.com/CartoDB/carto-vl/issues/950

### CODE REVIEW

File: docs/guidesdraft/01-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      style: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
+      center: [0, 30],
+      zoom: 2,
+      dragRotate: false
+    });
+```
+
+For basemaps you can add [Mapbox custom styles](https://www.mapbox.com/mapbox-gl-js/style-spec/) or choose one of the three predefined styles offered by CARTO:
+
+- **Voyager:** https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I'm currently changing those urls with new constants, so they will be replaced with `carto.basemaps.voyager` and alike

File: docs/guidesdraft/01-getting-started.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+## Getting started
+
+### Introduction
+
+[CARTO VL](https://github.com/cartodb/carto-vl) is a JavaScript library to create custom location intelligence applications that leverage the power of [CARTO](https://carto.com/). It uses [WebGL](https://www.khronos.org/webgl/) to enable powerful vector maps.
+
+It relies on [Mapbox GL](https://www.mapbox.com/mapbox-gl-js/api) to render the basemaps and it can be used for other functionality too, read [Mapbox GL documentation](https://www.mapbox.com/mapbox-gl-js/api/) for more information. However, keep in mind that CARTO VL layers can only be controlled with the CARTO VL API (this documentation) and that Mapbox GL native layers can only be controlled with the Mapbox GL API. Therefore, CARTO VL expressions cannot be used for Mapbox GL layers and vice versa.
+
+### Basic set up
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I think we should use "setup". [--> reference](http://grammarist.com/spelling/set-up-vs-setup/)

File: docs/guidesdraft/01-getting-started.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  <!-- Include Mapbox GL JS -->
+  <script src="https://libs.cartocdn.com/mapbox-gl/v0.48.0-carto1/mapbox-gl.js"></script>
+  <!-- Include Mapbox GL CSS -->
+  <link href="https://libs.cartocdn.com/mapbox-gl/v0.48.0-carto1/mapbox-gl.css" rel="stylesheet" />
+</head>
+```
+
+This will add the `carto` and the `mapboxgl` objects to the global namespace.
+
+> Note: Not every Mapbox GL version is compatible with CARTO VL. We highly recommend the following combination. If you are importing CARTO VL from npm, you have to use our Mapbox GL fork. Read more about how to do this in [the advanced guide]((https://carto.com/developers/carto-vl/guides/advanced))
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: Maybe we should add "Currently not..."

File: docs/guidesdraft/01-getting-started.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+## Getting started
+
+### Introduction
+
+[CARTO VL](https://github.com/cartodb/carto-vl) is a JavaScript library to create custom location intelligence applications that leverage the power of [CARTO](https://carto.com/). It uses [WebGL](https://www.khronos.org/webgl/) to enable powerful vector maps.
+
+It relies on [Mapbox GL](https://www.mapbox.com/mapbox-gl-js/api) to render the basemaps and it can be used for other functionality too, read [Mapbox GL documentation](https://www.mapbox.com/mapbox-gl-js/api/) for more information. However, keep in mind that CARTO VL layers can only be controlled with the CARTO VL API (this documentation) and that Mapbox GL native layers can only be controlled with the Mapbox GL API. Therefore, CARTO VL expressions cannot be used for Mapbox GL layers and vice versa.
+
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I miss an **introduction to the guide** itself, not the library. Something like, "In this guide, you will learn how to create a basic interactive map, displaying a dataset from CARTO on the top of a world basemap".  I think we should add this kind of intro to every guide.

Maybe just in this case, cause this is probably the entry point for many people, but I miss some kind of **expectations on the user**, like: "There are no previous requirements to complete this guide, but a basic understanding on HTML and JavaScript is helpful. You will need just a simple text editor and an Internet connection".

File: docs/guidesdraft/01-getting-started.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
+## Getting started
+
+### Introduction
+
+[CARTO VL](https://github.com/cartodb/carto-vl) is a JavaScript library to create custom location intelligence applications that leverage the power of [CARTO](https://carto.com/). It uses [WebGL](https://www.khronos.org/webgl/) to enable powerful vector maps.
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: Why a link to khronos, is it a business? Maybe we could point to a more recognized / neutral source like https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API ?

File: docs/guidesdraft/01-getting-started.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+It relies on [Mapbox GL](https://www.mapbox.com/mapbox-gl-js/api) to render the basemaps and it can be used for other functionality too, read [Mapbox GL documentation](https://www.mapbox.com/mapbox-gl-js/api/) for more information. However, keep in mind that CARTO VL layers can only be controlled with the CARTO VL API (this documentation) and that Mapbox GL native layers can only be controlled with the Mapbox GL API. Therefore, CARTO VL expressions cannot be used for Mapbox GL layers and vice versa.
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: IMO it is not very clear which is "the second it". I would rephrase to something like: "It relies on Mapbox GL to render the basemaps but that library can also be used to complement CARTO VL with some other functionalities (read Mapbox GL documentation for more information)"

File: docs/guidesdraft/01-getting-started.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+### Basic set up
+
+The easiest way to use CARTO VL is to include the required files from our CDN:
+
+* CARTO VL library: https://libs.cartocdn.com/carto-vl/%VERSION%/carto-vl.js
+* Minified file: https://libs.cartocdn.com/carto-vl/%VERSION%/carto-vl.min.js
+
+Then, we will need to add Mapbox GL JavaScript and CSS files.
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I would try to stick to the same "voice", using **you** will need and not *we*, for all the cases

File: docs/guidesdraft/01-getting-started.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    position: absolute;
+    height: 100%;
+    width: 100%;
+}
+```
+
+#### Defining the basemap
+
+Once we have the `div`, we use the `mapboxgl` object to initialize our map using the following parameters:
+
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I would add first the piece of js code and the explanation below

File: docs/guidesdraft/01-getting-started.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+- **Positron:** https://basemaps.cartocdn.com/gl/positron-gl-style/style.json
+- **Dark Matter:** https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json
+
+At this point you will have a basic map:
+
+[LiveExample](http://carto.com/developers/carto-vl/examples/maps/guides/getting-started/basic-set-up.html)
+
+### Defining the source
+
+In order to render data from CARTO you need to create a CARTO account and then get the necessary [credentials](https://carto.com/developers/fundamentals/authorization/).
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: Should we add some kind of link to the "create a CARTO account" part? 

I would also add a phrase explaining somehow that "you can use CARTO VL with different kind of sources, but it is easier and more powerful when connected to CARTO services" .

File: docs/guidesdraft/01-getting-started.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+The first thing you need to do is [authenticate the client](https://carto.com/developers/carto-vl/reference/#cartosetdefaultauth) with your `user` and `apiKey`.
+
+```js
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I would explain something like, "For this guide, you can use *cartovl* account and the next credentials, that give you access to `populated_places` dataset " (I think most people will do that), ..."but you can easily change them with yours to visualize your own datasets"

File: docs/guidesdraft/01-getting-started.md [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+The next step is to define the [`source`](https://carto.com/developers/carto-vl/reference/#cartosourcedataset) from your account to be displayed on the map. In the example below we are defining the `source` for a dataset named `populated_places` with all the populated places around the world.
+
+```js
+const source = new carto.source.Dataset('populated_places');
+```
+
+### Defining the Viz object
+
+A [`Viz object`](https://carto.com/developers/carto-vl/reference/#cartoviz) is one of the core elements of CARTO VL. It defines how the data will be styled, displayed, and processed. In this case we create an empty Viz object, that will use the default style.
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: "... the default style (a red circle)"

File: docs/guidesdraft/01-getting-started.md [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+```js
+const viz = new carto.Viz(`
+    color: red
+    width: 10
+`);
+```
+
+For more information about styling, check out the guide [Introduction to Styling](https://carto.com/developers/carto-vl/guides/introduction-to-styling/).
+
+### All together
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: Yes!! [All together now](https://www.youtube.com/watch?v=8jzDr--7E6o)

File: docs/guidesdraft/01-getting-started.md (hunk 4 above)
Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: YES! I have not thought about it but I think it is really useful and it is actually what you expect in an introduction ✨

File: docs/guidesdraft/01-getting-started.md (hunk 5 above)
Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: It is the official documentation afaik.

File: docs/guidesdraft/01-getting-started.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+  <!-- Include CARTO VL JS -->
+  <script src="https://libs.cartocdn.com/carto-vl/%VERSION%/carto-vl.min.js"></script>
+  <!-- Include Mapbox GL JS -->
+  <script src="https://libs.cartocdn.com/mapbox-gl/v0.48.0-carto1/mapbox-gl.js"></script>
+  <!-- Include Mapbox GL CSS -->
+  <link href="https://libs.cartocdn.com/mapbox-gl/v0.48.0-carto1/mapbox-gl.css" rel="stylesheet" />
+</head>
+```
+
+This will add the `carto` and the `mapboxgl` objects to the global namespace.
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: "global namespace" maybe can sound a little intimidating for non devs... 😄

File: examples/guides/getting-started/basic-styling.html [hunk 14]
```diff
@@ -41,15 +41,19 @@
...
         // Create the source
-        const source = new carto.source.Dataset('ne_10m_populated_places_simple');
+        const source = new carto.source.Dataset('populated_places');
+
         // Create a viz with some styles
         const viz = new carto.Viz(`
             color: red
             width: 10
         `);
+
```
Comment (VictorVelarde) [inline examples/guides/getting-started/basic-styling.html]: Amazing!!! I think I was just the only one bothered by no separation lines 😄

File: docs/guidesdraft/01-getting-started.md (hunk 9 above)
Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: I think we could add a link to the _sources_ section, where we talk about CARTO accounts a so on.

Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: Actually, there is a link to the sources but to the reference, maybe we can change it to the guides.

File: docs/guidesdraft/01-getting-started.md (hunk 11 above)
Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: This is not 100% right, I will think a better explanation of what default does, but I do not think it is needed at this point.

Comment (VictorVelarde) [reply docs/guidesdraft/01-getting-started.md]: ok, then let them see the default style alive 😄

File: docs/guidesdraft/01-getting-started.md [hunk 15]
```diff
@@ -0,0 +1,228 @@
+## Getting started
+
+### Introduction
+
+Welcome to the CARTO VL guides! This documentation is meant to lead you from the basics of creating a map to advanced techniques for developing powerful interactive visualizations.
+
+In this guide, you will learn how to visualize information with CARTO VL by displaying a dataset on the top of a world basemap. If there is any word or term you do not understand while reading the guides, please take a look to our [Glossary](https://carto.com/developers/carto-vl/guides/glossary).
+
```
Comment (elenatorro) [inline docs/guidesdraft/01-getting-started.md]: I know that `https://carto.com/developers/carto-vl/guides/glossary` is not a valid link right now, but we will review all the links on the fly.

File: docs/guidesdraft/01-getting-started.md [hunk 16]
```diff
@@ -0,0 +1,228 @@
...
+```css
+#map {
+    position: absolute;
+    height: 100%;
+    width: 100%;
+}
+```
+
+#### Defining the basemap
```
Comment (elenatorro) [inline docs/guidesdraft/01-getting-started.md]: Change all the headers to follow the same structure as in the examples https://github.com/CartoDB/carto-vl/issues/966

File: docs/guidesdraft/01-getting-started.md (hunk 15 above)
Comment (VictorVelarde) [reply docs/guidesdraft/01-getting-started.md]: @elenatorro could we link to specific terms in that glossary? Like https://carto.com/developers/carto-vl/guides/glossary#GIS

That would be very useful!

Comment (elenatorro) [reply docs/guidesdraft/01-getting-started.md]: Totally! There is an open issue for the glossary https://github.com/CartoDB/carto-vl/issues/979, we will do that for sure :)

File: docs/guidesdraft/01-getting-started.md [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+
+### What is CARTO VL?
+
+[CARTO VL](https://github.com/cartodb/carto-vl) is a JavaScript library to create custom location intelligence applications that leverage the power of [CARTO](https://carto.com/). It uses [WebGL](https://www.khronos.org/webgl/) to enable powerful vector maps.
+
+It relies on [Mapbox GL](https://www.mapbox.com/mapbox-gl-js/api) to render the basemaps. Mapbox GL can be used for other functionality too, read [Mapbox GL documentation](https://www.mapbox.com/mapbox-gl-js/api/) for more information. However, keep in mind that CARTO VL layers can only be controlled with the [CARTO VL API]() and that Mapbox GL native layers can **only** be controlled with the Mapbox GL API. Therefore, CARTO VL expressions cannot be used for Mapbox GL layers and vice versa.
+
+### Basic setup
+
+<div class="example-map">
```
Comment (VictorVelarde) [inline docs/guidesdraft/01-getting-started.md]: I'm not sure if this is the place for the end result iframe, is it? Should it be also preceded by a sentence like: "This is the final result of this guide" or something like that?

**Comment by VictorVelarde** [comment]
It LGTM, there is just a constant to add at `basic-setup.html` instead of the url for the style. but the PR looks pretty good! 👍

**Comment by VictorVelarde** [comment]
After that I think we can merge it!

**Comment by elenatorro** [comment]
Hi @VictorVelarde, I updated the examples to `step-1` and `step-2` to follow the convention :) However, I think that dealing with numbers helps us to know the guides order when looking at the code immediately. It is true that if we do that, then we would have to change it if we want to include one guide in between other two, but I do not think it is going to happen very often once we release the public version, and if so there is no problem with renaming the files in my opinion.

**Comment by VictorVelarde** [comment]
Ok. Then I'll rename mine, adding the '02-' preffix in this very same PR, fix the link from 01-getting-started, and then merge to master
